### PR TITLE
WIP: Adding RoaringBitmap to compress the bitmap in cidr_set

### LINKED ISSUE #44918: cidr_set bitmap should be compressed
**Is this a request for help?** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.):

**What keywords did you search in Kubernetes issues before filing this one?** (If you have found any duplicates, you should instead reply there.):

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

**Kubernetes version** (use `kubectl version`):
latest

**Environment**:
- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release):
- **Kernel** (e.g. `uname -a`):
- **Install tools**:
- **Others**:

**What happened**:
cidr_set.go in pkg/controller/node currently uses a bitmap to track the subnets allocated.
The current allocation logic assumes a small allocation range. This bitmap could grow very large with the introduction of IPv6.
The v6 code in cidr_set.go currently limits the difference between the cluster mask size and subnet mask size to 16. This restriction could be removed if the bitmap was compressed.

**What you expected to happen**:

**How to reproduce it** (as minimally and precisely as possible):

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by rpothier**
**What this PR does / why we need it**:

This is a WIP, to add bitmap compression cidr_set.
This is needed with the addition of IPv6, the cidr_set bitmap can get very large.
Pushing as a WIP due to:
   gather comments from the community,
   godep seems to have added files in k8s.io/metrics, should that be there or in vendor?
   Need to set a version as described in https://github.com/kubernetes/kubernetes/issues/44918#issuecomment-301184034

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #44918

**Special notes for your reviewer**:

**Release note**:

```NONE
```

### CODE REVIEW

**Comment by smarterclayton** [comment]
Will have to wait a bit before I have time to look.

File: Godeps/LICENSES [hunk 1]
```diff
@@ -35282,6 +35282,32 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 
 ================================================================================
+= vendor/github.com/glycerine/go-unsnap-stream licensed under: =
+
+Copyright (c) 2014 the go-unsnap-stream authors.
```
Comment (thockin) [inline Godeps/LICENSES]: This is not a standard license.  I'll reach out to the authors, but I am not sure we can use it.

**Review by thockin — COMMENTED** [review]
needs rebase

**Comment by dims** [comment]
Looks like they started from MIT license and removed changed it - https://github.com/glycerine/go-unsnap-stream/commit/ba4a6d32d957f52defb0caad0b1a936126d31858#diff-9879d6db96fd29134fc802214163b95a

**Comment by smarterclayton** [comment]
Looks like BSD but without the two clauses.

**Comment by thockin** [comment]
Assign back to us when ready to review.

File: Godeps/LICENSES (hunk 1 above)
Comment (rpothier) [reply Godeps/LICENSES]: @thockin Were you able to determine anything? Can this license be used? or do I need to look for an alternative compression method.

**Pull request closed without merging**